src/tox/session/env_select.py

```diff
@@ -320,14 +330,19 @@ def iter(
 
         :return: an iteration of tox environments
         """
-        ignore_envs: set[str] = set()
+        tox_env_filter = os.environ.get("TOX_SKIP_ENV")
+        tox_env_filter_re = re.compile(tox_env_filter) if tox_env_filter is not None else None
```

Comment:
This definition should happen in the constructor of this class, not every time on this call, no?